src/main/java/fr/openent/competences/controllers/ExportPDFController.java

```diff
@@ -259,30 +286,45 @@ private void saveBfcWorker(List<String> idClasses, Long idCycle, JsonObject resu
                     JsonObject body = message.body();
+                    //ICI insérer dans le mongo
+
                     if(body.getJsonObject("result").getJsonObject("c").
                             getJsonObject("metadata").getJsonArray("labels").contains("Class")){
                         result.put("idClasse" , idClasses.get(0));
-                        JsonObject action = new JsonObject().put(ACTION, BulletinWorker.SAVE_BFC)
-                                .put("request", jsonRequest)
-                                .put("resultFinal", result)
-                                .put("template", templateName)
-                                .put("title", prefixPdfName);
-
-                        eb.send(BulletinWorker.class.getSimpleName(), action, Competences.DELIVERY_OPTIONS);
+//                        JsonObject action = new JsonObject().put(ACTION, BulletinWorker.SAVE_BFC)
+//                                .put("request", jsonRequest)
+//                                .put("resultFinal", result)
+//                                .put("template", templateName)
+//                                .put("title", prefixPdfName);
+
+                        setFuturesToInsertMongo(result, prefixPdfName, templateName, jsonRequest);
+//                        eb.send(BulletinWorker.class.getSimpleName(), action, Competences.DELIVERY_OPTIONS);
                     }
                 }));
             }else{
-                JsonObject action = new JsonObject().put(ACTION, BulletinWorker.SAVE_BFC)
-                        .put("request", jsonRequest)
-                        .put("resultFinal", result)
-                        .put("template", templateName)
-                        .put("title", prefixPdfName);
-
-                eb.send(BulletinWorker.class.getSimpleName(), action, Competences.DELIVERY_OPTIONS);
+//                JsonObject action = new JsonObject().put(ACTION, BulletinWorker.SAVE_BFC)
+//                        .put("request", jsonRequest)
+//                        .put("resultFinal", result)
+//                        .put("template", templateName)
+//                        .put("title", prefixPdfName);
+                setFuturesToInsertMongo(result, prefixPdfName, templateName, jsonRequest);
+//                eb.send(BulletinWorker.class.getSimpleName(), action, Competences.DELIVERY_OPTIONS);
             }
+
         }
     }
 
+    private void setFuturesToInsertMongo(JsonObject result, String prefixPdfName, String templateName, JsonObject jsonRequest) {
+        List<Future<String>> futureArray = insertDataInMongo(result,jsonRequest,prefixPdfName,templateName);
+        FutureHelper.all(futureArray).onSuccess(success ->{
+            log.info("insert bfc data in Mongo done");
+            eb.send(BulletinWorker.class.getSimpleName(), new JsonObject(), Competences.DELIVERY_OPTIONS);
+            //ping le worker
+        }).onFailure(error ->{
+            log.info(error.getMessage());
```

Comment:
Comme ça sera dans les log de vertx, est-ce qu'il ne serait pas plus pertinent d'envoyer des messages de log + parlant ?
Par exemple utiliser :
`log.info(String.format("[Competences@%s::setFuturesToInsertMongo] an error has occurred during insert data in mongo: %s.", this.getClass().getSimpleName(), error.getMessage()));`

Reply:
tu peux faire log.error aussi